### ISSUE TITLE
.NET 4.7.1 console app tries to use BenchmarkRunner, gets "Could not load file or assembly 'System.Runtime, Version=4.1.2.0"

### ISSUE THREAD

**Issue by sandersaares** [dotnet/BenchmarkDotNet#896]
1. Create a new .NET Framework 4.7.1 console app project.
1. Install BenchmarkDotNet NuGet package with DependencyBehavior=Highest to get non-broken dependencies.
1. Paste the following code into Program.cs and excute the app.

```cs
using BenchmarkDotNet.Attributes;
using BenchmarkDotNet.Running;

namespace ConsoleApp3
{
    internal class Program
    {
        private static void Main(string[] args)
        {
            BenchmarkRunner.Run<Foo>();
        }
    }

    [ClrJob]
    public class Foo
    {
        [Benchmark]
        public void Bar()
        {

        }
    }
}
```

Expected result: app executes and does some no-op benchmarks.

Actual result:

```
// Validating benchmarks:
// ***** BenchmarkRunner: Start   *****
// ***** Building 1 exe(s) in Parallel: Start   *****
// ***** Done, took 00:00:00 (0.13 sec)   *****
// Found benchmarks:
//   Foo.Bar: Clr(Runtime=Clr)

// Build Exception: Could not load file or assembly 'System.Runtime, Version=4.1.2.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.
```

**Comment by adamsitnik** [comment]
Hi @sandersaares 

I was able to reproduce the problem. VS generates invalid assembly binding redirects.

The solution is to remove the assembly binding redirects from the project.

I am going to close it since it's not a BDN issue.

**Comment by sandersaares** [comment]
Using BDN with a very simple app does not work - I think this is a BDN issue no matter what causes it. Do you expect users to start debugging Visual Studio or BDN to make benchmarks? I mean, what is the path forward here? All you are saying to me right now is that I cannot use BDN.

**Comment by adamsitnik** [comment]
BenchmarkDotNet has correct dependencies and a very simple `.nuspec` file. If some 3rd party tool can't use it correctly it's not our fault.

As I said all you need to do is just to remove the assembly binding redirects and it's going to work.  This is a known issue and the workaround works https://github.com/dotnet/BenchmarkDotNet/issues/667 

I know it's a workaround but this is an Open Source project and both me and @AndreyAkinshin maintain it in our free time, we don't have time to dig into every issue and provide a perfect solution if there is a known workaround. If you feel like it's wrong you can fork the code and try to fix the issue. I would be very happy to merge a PR with fix.

**Comment by sandersaares** [comment]
.NET Framework is not "some 3rd party tool" and BDN not working with .NET 4.7.1 is a major issue that needs to be fixed in order to have this library be usable.

Sure, I understand you might not have the availability to fix it. But I ask that you at least keep valid issues open.

**Comment by sandersaares** [comment]
What exactly are these "invalid binding redirects"? I request a link to the bug report about them that has been sent to Microsoft (I assume one is submitted since this sounds like a known issue).

**Comment by paolofulgoni** [comment]
@sandersaares  maybe it is caused by this project setting:
`[X] Auto-generate binding redirects`

**Comment by adamsitnik** [comment]
This was one big PITA, fixed by using some ugly workarounds in #1032
